src/Samples/SubModelSeq/Program.fs

```diff
@@ -13,177 +31,199 @@ module List =
-        CounterValue = 0
+    let init =
+      { Count = 0
         StepSize = 1 }
+    
+    let canReset = (<>) init
+    
+    let update msg m =
+      match msg with
+      | Increment -> { m with Count = m.Count + m.StepSize }
+      | Decrement -> { m with Count = m.Count - m.StepSize }
+      | SetStepSize x -> { m with StepSize = x }
+      | Reset -> init
 
-    let increment c =
-      { c with CounterValue = c.CounterValue + c.StepSize }
-
-    let decrement c =
-      { c with CounterValue = c.CounterValue - c.StepSize}
-
-    let setStepSize step c =
-      { c with StepSize = step }
 
-    let canReset c =
-      c.CounterValue <> 0 || c.StepSize <> 1
 
-    let reset c =
-      { create () with Id = c.Id }
+[<AutoOpen>]
+module TreeTypes =
 
+  type RoseTree<'a> =
+    { Data: 'a
+      Children: RoseTree<'a> list }
 
-module Tree =
 
-  type Node<'a> =
-    { Data: 'a
-      Children: Node<'a> list }
+  module RoseTree =
 
-  let asLeaf a =
-    { Data = a
-      Children = [] }
+    module Data =
+      let get m = m.Data
+      let set v m = { Data = v; Children = m.Children }
+      let rec map f n =
+        { Data = n.Data |> f
+          Children = n.Children |> (f |> map |> List.map) }
+    module Children =
+      let get m = m.Children
+      let set v m = { m with Children = v }
+      let map f = map get set f
 
-  let dataOfChildren n =
-    n.Children |> List.map (fun nn -> nn.Data)
+    let rec map f t =
+      t |> f |> Children.map (f |> map |> List.map)
 
-  let rec map f n =
-    let nn = n |> f
-    { nn with Children = nn.Children |> List.map (map f) }
+    let asLeaf a =
+      { Data = a
+        Children = [] }
 
-  let rec mapData f n =
-    { Data = n.Data |> f
-      Children = n.Children |> List.map (mapData f) }
+    let addChild a = a |> asLeaf |> List.cons |> Children.map
+    
+    let dataOfChildren n =
+      n.Children |> List.map Data.get
+    
+    let rec preorderFlatten n =
+      n :: List.collect preorderFlatten n.Children
 
-  let rec preorderFlatten n =
-    n :: List.collect preorderFlatten n.Children
 
 module App =
 
   type Model =
     { SomeGlobalState: bool
-      DummyRoot: Tree.Node<Counter> }
+      DummyRoot: RoseTree<Identifiable<Counter>> }
+
+  type Msg =
+    | ToggleGlobalState
+    | AddCounter of parent: Guid
+    | CounterMsg of Guid * CounterMsg
+    | Remove of Guid
+    | MoveUp of Guid
+    | MoveDown of Guid
+
+
+  module SomeGlobalState =
+    let get m = m.SomeGlobalState
+    let set v m = { m with SomeGlobalState = v }
+    let map f = f |> map get set
+  module DummyRoot =
+    let get m = m.DummyRoot
+    let set v m = { m with DummyRoot = v }
+    let map f = f |> map get set
 
-  let asDummyRoot ns : Tree.Node<Counter> =
-    { Data = Counter.create() // Placeholder data to satisfy type system. User never sees this.
-      Children = ns }
+  let createIdentifiableCounter () =
+    () |> Guid.NewGuid |> Identifiable.create Counter.init
```

Comment:
I don't like this kind of "pipe-o-mania".

```suggestion
    { Id = Guid.NewGuid ()
      Value = Counter.init }
```

Reply:
Sure.  I made this change.